File: 1608/1608.go
func specialArrayWithXElementsGreat(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := specialArrayWithXElementsGreat(root.Left)
	r := specialArrayWithXElementsGreat(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 1608/1608.cs
public class Solution {
    public int SpecialArrayWithXElementsGreat(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(SpecialArrayWithXElementsGreat(root.left), SpecialArrayWithXElementsGreat(root.right));
    }
}